frontend/guide: update buy guide content

Beerosagos opened the pull request:
The guide content for the buy section is updated to cover the integration of Pocket.

thisconnect commented on frontends/web/src/locales/en/app.json:
```diff
@@ -372,13 +373,12 @@
         },
         "security": {
-          "description": "When you buy {{name}} via MoonPay, you are using an external service. This service is out of scope of the BitBox02 Security Threat model and relies on the safety and security of the environment which the BitBoxApp software is running in.",
+          "description": "When you buy {{name}} via MoonPay and/or Pocket, you are using an external service. This service is out of scope of the BitBox02 Security Threat model and relies on the safety and security of the environment which the BitBoxApp software is running in.",
```
since you are here anyway, please lowercase "Security Threat model"

thisconnect replied:
One more thing, updating existing text should be done in locize. but keep it here for now so we see what should change and can discuss the change.

once PR is accepted we can change locize.

Beerosagos replied:
I changed "Security Threat model", but restored the rest of the text as it was in the beginning. Made a new `descriptionGeneric` with a generic text that avoid mentioning single exchanges, which is included in the guide.

thisconnect commented on frontends/web/src/routes/buy/guide.tsx:
```diff
@@ -36,9 +36,13 @@ export default function BuyGuide({ name }: BuyGuideProps) {
           url: 'https://www.moonpay.com/privacy_policy',
         },
+        link2: {
+          text: t('buy.pocket.data.link'),
```
I would expect a similar key like moonpayPrivacyPolicy, but I expect it is probably due to that this content is shared...

on the other hand iirc it is also shared for moonpay. Could you double check why this key is not `buy.info.disclaimer.pocketPrivacyPolicy`... or why the moonpay one isn't using 'buy.moonpay.data.link'  

just fyi, we shouldn't rename existing keys, so if possible it would be great to have consistent keys.

Beerosagos replied:
Yes, the content is shared with the terms and conditions. Since we would need to refactor also the keys related to moonpay, but decided not to do it now, I didn't change it.

thisconnect commented on frontends/web/src/routes/buy/guide.tsx:
```diff
@@ -36,9 +36,13 @@ export default function BuyGuide({ name }: BuyGuideProps) {
         },
+        link2: {
+          text: t('buy.pocket.data.link'),
+          url: 'https://pocketbitcoin.com/policy/privacy',
```
nit: could add https://pocketbitcoin.com/de/policy/privacy and https://pocketbitcoin.com/it/policy/privacy (German is fine but Italian seems to not be translated yet).

moonpay is also hardcoded to EN, so feel free to keep as is.

Beerosagos replied:
I think it is better to keep this PR simple and make a separated PR where we call localized policies for both moonpay and pocket. wdyt?

thisconnect commented on frontends/web/src/routes/buy/moonpay-terms.tsx:
```diff
@@ -94,7 +94,7 @@ export const MoonpayTerms = ({ account, onAgreedTerms }: TProps) => {
         <p>
           <A className={style.link} href="https://www.moonpay.com/privacy_policy">
-            {t('buy.info.disclaimer.privacyPolicy')}
+            {t('buy.info.disclaimer.moonpayPrivacyPolicy')}
```
aha, related to my other comment.. since you rename anyway maybe make it consistent with the key of the pocket link?

Beerosagos replied:
Restored the previous name to avoid modifying keys

thisconnect commented on frontends/web/src/locales/en/app.json:
```diff
@@ -372,13 +373,12 @@
-        "privacyPolicy": "MoonPay privacy policy",
         "protection": {
-          "description": "The BitBoxApp does not collect any data when buying {{name}}, the incoming funds are treated like a regular transaction. MoonPay needs to collect some personal data to operate. Their Privacy Policy explains in detail how that data is handled.",
+          "description": "The BitBoxApp does not collect any data when buying {{name}}, the incoming funds are treated like a regular transaction. However MoonPay and Pocket need to collect some information to operate. Please refer to their respective privacy policies to see in more detail how the data is handled.",
```
I think I am missing context, could you update the commit message with how this should work?

Is the following correct?

each exchange/broker terms has its own text, followed by a shared text (this one). which is also shared in the guide.

I am confused by that in the "own" text it's the exchange/broker name, but in the next paragraph it suddently mentioned other service that I didn't select. This would be less confusing if we just used generic term instead of the actual name.

Also what if we add 5 more exchanges broker one day, maybe the user doesn't even know about them if they are not available in different regions. Then the text lists them all?

ping @jadzeidan 

> However MoonPay, Pocket, Broker2, Exchange3, BrokerNotInMyRegion4 or P2PXchange need to collect..

thisconnect commented on frontends/web/src/components/guide/entry.tsx:
```diff
@@ -57,6 +61,9 @@ export const Entry: FunctionComponent<TProps> = props => {
             )}
+            {entry.link2 && (
+              <p><A data-testid="link" href={entry.link2.url}>{entry.link2.text}</A></p>
+            )}
```
what if there are 5 more exchanges/brokers?

nicer IMO would be if the exchange/broker selection view would be generic (no link). and once the user selected a service the correct one would show up.

Beerosagos replied:
Good point, now there is an optional parameter that allows linking the correct privacy policy, or no privacy policy at all if the exchange is not defined.

thisconnect reviewed (changes requested):
was confused about shared text/links

Beerosagos commented:
Thanks for the review! I pushed an update: 
- tried to avoid modifying locize keys or texts (exception for "Security Thread model" to lowercase, as you requested)
- added generic privacy/security text keys to be included in the guide, which do not specify any exchange name
- added an optional parameter in the guide, to handle privacy links
- Bonus: fixed a small HTML mistake in pocket terms. Is a bit OT, but hope it is fine

PTAL :pray:

thisconnect commented on frontends/web/src/locales/en/app.json:
```diff
@@ -375,10 +375,12 @@
         },
         "security": {
-          "description": "When you buy {{name}} via MoonPay, you are using an external service. This service is out of scope of the BitBox02 Security Threat model and relies on the safety and security of the environment which the BitBoxApp software is running in.",
+          "description": "When you buy {{name}} via MoonPay, you are using an external service. This service is out of scope of the BitBox02 security threat model and relies on the safety and security of the environment which the BitBoxApp software is running in.",
```
thanks 👍 
I also lowercased "security thread" in locize

thisconnect reviewed (approved):
👍  LGTM tested
